script.php

```diff
@@ -0,0 +1,220 @@
+            } elseif (isset($definition['example']['type'])) {
+                $modelTypeValue = $definition['example']['type'];
+            } else {
+                $modelTypeValue = strtolower($model);
+            }
+
+            if (! $modelType || ! $modelTypeValue || ! $properties) {
+                continue;
+            }
+            $requiredProperties = $definition['properties']['attributes']['required'] ?? [];
+
+            $content .= "    static type: {$modelType} = '{$modelTypeValue}';" . PHP_EOL . PHP_EOL;
+
+            foreach ($properties as $propertyName => $propertyInfo) {
+                if (! isset($propertyInfo['x-module']) || $propertyInfo['x-module'] !== $modelModule) {
+                    continue;
+                }
+
+                [$propertyType, $addImport] = $this->getPropertyType($propertyInfo);
+                $additionalImport = array_merge($additionalImport, $addImport);
+
+                if (! isset($propertyInfo['default']) && in_array($propertyName, $requiredProperties)) {
+                    $defaultValue = '';
+                } elseif (isset($propertyInfo['default'])) {
+                    $defaultValue = 'string' === $propertyType
+                        ? " = '{$propertyInfo['default']}'"
+                        : " = {$propertyInfo['default']}";
+                } else {
+                    $defaultValue = ' = null';
+                }
+
+                $content .= "    {$propertyName}: {$propertyType}";
+                $content .= in_array($propertyName, $requiredProperties) ? '' : ' | null';
+                $content .= $defaultValue;
+                $content .= ';' . PHP_EOL;
+            }
+
+            $content = str_replace(
+                '*additionalImport*' . ($additionalImport ? '' : PHP_EOL . PHP_EOL),
+                implode(PHP_EOL . PHP_EOL, $additionalImport),
+                $content
+            );
+
+            $content .= PHP_EOL;
+            $content .= '    constructor(config: any) {' . PHP_EOL;
+            $content .= '        super(config);' . PHP_EOL . PHP_EOL;
+            $content .= '        this.collectAttributes(config);' . PHP_EOL;
+            $content .= '    }' . PHP_EOL;
+            $content .= '}';
+
+            file_put_contents("output/{$model}Model.ts", $content);
+        }
+    }
+
+    protected function getPropertyType(array $propertyInfo): array
+    {
+        $additionalImport = [];
+        if (isset($propertyInfo['type'])) {
+            $propertyType = match ($propertyInfo['type']) {
+                'integer' => 'number',
+                default => $propertyInfo['type'],
+            };
+        } elseif(isset($propertyInfo['$ref'])) {
+            $pos = strrpos($propertyInfo['$ref'], '/');
+            $propertyType = $pos !== false ? substr($propertyInfo['$ref'], $pos + 1) : $propertyInfo['$ref'];
+            $additionalImport[$propertyType] = $this->addImportRefType($propertyType);
```

Comment:
add map of refs which should be imported as objects to config,
in case you have different ref here - print warning that it's not found